packit/upstream.py

```diff
@@ -951,26 +712,398 @@ def convert_version_to_tag(self, version_: str) -> str:
+        except IndexError:
+            raise PackitSRPMNotFoundException(
+                "SRPM cannot be found, something is wrong."
+            )
+        return the_srpm
+
+    def get_build_command(self) -> Tuple[List[str], str]:
+        """
+        Constructs `rpmbuild` command.
+
+        Returns:
+            Escaped `rpmbuild` command.
+        """
+        cmd = [
+            "rpmbuild",
+            "-bs",
+            "--define",
+            f"_sourcedir {self.rpmbuild_dir}",
+            "--define",
+            f"_srcdir {self.rpmbuild_dir}",
+            "--define",
+            f"_specdir {self.rpmbuild_dir}",
+            "--define",
+            f"_srcrpmdir {self.srpm_dir}",
+            "--define",
+            f"_topdir {self.rpmbuild_dir}",
+            # we also need these 3 so that rpmbuild won't create them
+            "--define",
+            f"_builddir {self.rpmbuild_dir}",
+            "--define",
+            f"_rpmdir {self.rpmbuild_dir}",
+            "--define",
+            f"_buildrootdir {self.rpmbuild_dir}",
+            self.upstream.package_config.specfile_path,
+        ]
+        escaped_command = " ".join(cmd)
+
+        logger.debug(f"SRPM build command: {escaped_command}")
+        return cmd, escaped_command
+
+    def get_path(self, out: str) -> Path:
+        """
+        Get path to the SRPM file. In case it is necessary to move the built SRPM,
+        move it.
+
+        Args:
+            out (str): Output of the `rpmbuild` command.
+
+        Returns:
+            Path to the SRPM.
+        """
+        built_srpm_path = self._get_srpm_from_rpmbuild_output(out)
+        if self.srpm_path:
+            shutil.move(built_srpm_path, self.srpm_path)
+            return Path(self.srpm_path)
+
+        if self.upstream.running_in_service():
+            return self.upstream.local_project.working_dir / built_srpm_path
+        return Path(built_srpm_path)
+
+    def build(self) -> Path:
+        cmd, escaped_command = self.get_build_command()
+
+        present_srpms = set(self.srpm_dir.glob("*.src.rpm"))
+        logger.debug(f"Present SRPMs: {present_srpms}")
+
+        try:
+            out = self.upstream.command_handler.run_command(
+                cmd, return_output=True
+            ).strip()
+        except PackitCommandFailedError as ex:
+            logger.error(f"The `rpmbuild` command failed: {ex!r}")
+            raise PackitFailedToCreateSRPMException(
+                f"reason:\n{ex}\n"
+                f"command:\n{escaped_command}\n"
+                f"stdout:\n{ex.stdout_output}\n"
+                f"stderr:\n{ex.stderr_output}"
+            ) from ex
+        except PackitException as ex:
+            logger.error(f"The `rpmbuild` command failed: {ex!r}")
+            raise PackitFailedToCreateSRPMException(
+                f"The `rpmbuild` command failed:\n{ex}"
+            ) from ex
+
+        return self.get_path(out)
+
+
+class PrepareForSRPM:
```

Comment:
I like when classes use names so one can tell what's inside - this one doesn't do it for me :/

would it make sense to merge it with the builder above?

I understand the methods here are pretty random but this refactoring PR could be a great opportunity to clean things up.